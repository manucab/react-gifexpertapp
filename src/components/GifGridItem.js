import React from 'react'

const GifGridItem = ( {id, title, url} ) => {
    

    return (
        <div className='card animate_animated animate__fadeIn'>
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
}

export default GifGridItem
