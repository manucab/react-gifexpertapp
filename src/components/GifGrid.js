import React from "react";
import { useFetchGifs } from "../Hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data:images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate_animated animate__fadeIn'> {category} </h3>

      {loading && <p className='animate_animated animate__flash'>Cargando</p>}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
