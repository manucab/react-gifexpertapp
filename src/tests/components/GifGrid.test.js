import React from 'react'
import '@testing-library/jest-dom'

import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../Hooks/useFetchGifs';
jest.mock('../../Hooks/useFetchGifs')

describe("Pruebas en GifGrid", () => {
  
    const category = "Goku";


  test("debe devolver el componente correctamente", () => {

    useFetchGifs.mockReturnValue({
        data:[],
        loading:true
    })
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

    const gifs = [{
        id: 'ABC',
        url:'https://www.google.es',
        title:'Cualquier cosa'
    }]

    useFetchGifs.mockReturnValue({
        data: gifs,
        loading:false
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect( wrapper ).toMatchSnapshot()
    expect( wrapper.find('p').exists() ).toBe(false)
    //Buscamos por nombre del elemento
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )

  })
  


});
