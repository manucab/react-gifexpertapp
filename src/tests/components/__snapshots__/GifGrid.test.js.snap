// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate_animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://www.google.es"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid debe devolver el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate_animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <p
    className="animate_animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
